tt_metal/impl/dispatch/device_command.hpp

```diff
@@ -249,10 +248,24 @@ class DeviceCommand {
             initialize_write_cmd(write_cmd_dst);
         }
 
-        if constexpr (inline_data) {
-            TT_ASSERT(data != nullptr);  // compiled out?
-            uint32_t increment_sizeB = align(data_sizeB, this->pcie_alignment);
-            this->add_data(data, data_sizeB, increment_sizeB);
+        // Case 1: flush_prefetch
+        //  a) there is inline_data: data is provided here and follows prefetch relay inline and cq dispatch write linear so total increment size is:
+        //          align(sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd) + data_sizeB, pcie_alignment)
+        //  b) don't have inline_data: next command should be to add_data (don't do aligned increment) so total increment size is:
+        //          sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd)
+        // Case 2: !flush_prefetch: no data, increment size is:
+        //          align(sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd), pcie_alignment)
+        // Note that adding prefetch_relay_inline and writing the dispatch command already increment cmd_write_offsetB via calls to reserve_space
+        if constexpr (flush_prefetch) {
+            if constexpr (inline_data) {
+                TT_ASSERT(data != nullptr);  // compiled out?
+                uint32_t increment_sizeB = align(data_sizeB, this->pcie_alignment);
```

Comment:
This still needs to be fixed right?
The total inc calculated for this case would be `sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd) + align(data_sizeB, this->pcie_alignment)`, but `sizeof(CQPrefetchCmd) + sizeof(CQDispatchCmd)` should be part of the alignment.